src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQueryAvroToStructuredTransformer.java

```diff
@@ -76,6 +77,11 @@ protected Object convertField(Object field, Schema fieldSchema) throws IOExcepti
           case TIMESTAMP_MILLIS:
           case TIMESTAMP_MICROS:
             return field;
+          case DECIMAL:
+            ByteBuffer value = (ByteBuffer) field;
+            byte[] bytes = new byte[value.remaining()];
+            value.get(bytes);
```

Comment:
need to reset the position in the ByteBuffer, otherwise something else that uses it will get messed up:

```
    int pos = value.position();
    value.get(bytes);
    value.position(pos);
```